environment/src/main/java/com/sequenceiq/environment/environment/service/sdx/SdxUpgradeCcmPollerService.java

```diff
@@ -0,0 +1,36 @@
+package com.sequenceiq.environment.environment.service.sdx;
+
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import com.dyngr.Polling;
+import com.sequenceiq.environment.environment.poller.SdxPollerProvider;
+
+@Service
+public class SdxUpgradeCcmPollerService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SdxUpgradeCcmPollerService.class);
+
+    @Value("${env.upgradeccm.datalake.polling.attempt:45}")
+    private Integer attempt;
+
+    @Value("${env.upgradeccm.datalake.polling.sleep.time:20}")
+    private Integer sleeptime;
+
+    private final SdxPollerProvider pollerProvider;
+
+    public SdxUpgradeCcmPollerService(SdxPollerProvider pollerProvider) {
+        this.pollerProvider = pollerProvider;
+    }
+
+    public void waitForUpgradeCcm(Long envId, String datalakeCrn) {
+        Polling.stopAfterAttempt(attempt)
+                .stopIfException(true)
+                .waitPeriodly(sleeptime, TimeUnit.SECONDS)
+                .run(() -> pollerProvider.upgradeCcmPoller(envId, datalakeCrn));
+    }
```

Comment:
Possibly wrap inside `try {} catch (PollerStoppedException e)` like in case of `FreeIpaPollerService.waitForCcmUpgrade()`.

Also, not a big deal, but the method naming is not consistent (`UpgradeCcm` vs `CcmUpgrade`). :smile: